src/controllers/funds.js

```diff
@@ -0,0 +1,32 @@
+const { default: mongoose } = require('mongoose');
+const Funds = require('../models/fund');
+
+module.exports = {
+    getFunds: async (req, res) => {
+        try {
+            const { category, baseUserId, lastDate, currentDate } = req.query;
+            const filter = {};
+            if (category) {
+                filter.category = { $in: category };
+            }
+            if (baseUserId) {
+                filter.publisherId = mongoose.Types.ObjectId(baseUserId);
+            }
+            if (lastDate && currentDate) {
+                filter.createdAt = { $gte: currentDate, $lte: lastDate };
+            }
+
+            const filteredItem = await Funds.find(filter);
+            if (!filteredItem) {
+                return res.status(400).json({
+                    error: 'enter a valid query!',
+                });
+            }
+            res.status(200);
+            res.json(filteredItem);
+        } catch (err) {
+            res.json(err.msg);
```

Comment:
Don't send the error message to the user like this, it might contain sensitive data. If you have an error that you don't want to handle just send `return res.sendStatus(500);`